internal/v1/server.go

```diff
@@ -158,41 +154,60 @@ func Attach(conf *ServerConfig) (*Server, error) {
 				next.ServeHTTP(w, r)
 			}
 		})
-	}))
-
-	// Global middleware: extract or generate trace information
-	s.echo.Pre(strc.NewEchoV4MiddlewareWithConfig(slog.Default(), strc.MiddlewareConfig{
-		Filters: []strc.Filter{
-			strc.IgnorePathPrefix("/metrics"),
-			strc.IgnorePathPrefix("/status"),
-			strc.IgnorePathPrefix("/ready"),
-		},
-	}))
-
-	// Global middleware: log stack traces into standard logger as error (instead of stdout)
-	s.echo.Use(middleware.RecoverWithConfig(middleware.RecoverConfig{
+	})
+
+	// Middleware: enforce identity presence for authenticated routes
+	enforceIdentityMiddleware := func(next echo.HandlerFunc) echo.HandlerFunc {
+		return func(c echo.Context) error {
+			if id := c.Request().Context().Value(fedora_identity.IDHeaderKey); id == nil {
+				return echo.NewHTTPError(http.StatusBadRequest, "Missing identity header")
+			}
+			return next(c)
+		}
+	}
+
+	// Middleware: log stack traces into standard logger as error (instead of stdout)
+	recoverMiddleware := middleware.RecoverWithConfig(middleware.RecoverConfig{
 		LogLevel: glog.ERROR,
-	}))
+	})
 
-	// Route-based middleware:
-	var middlewares []echo.MiddlewareFunc
+	// Middleware: structured logging and tracing
+	// XXX: split into individual logging and tracing middlewares
```

Comment:
Here I like the comment because the `NewEchoV4MiddlewareWithConfig` name isn't clear.